test/speech/index.js

```diff
@@ -0,0 +1,836 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var assert = require('assert');
+var async = require('async');
+var extend = require('extend');
+var fs = require('fs');
+var path = require('path');
+var mockery = require('mockery-next');
+var nodeutil = require('util');
+var EventEmitter = require('events');
+var tmp = require('tmp');
+
+var Service = require('../../lib/common/service.js');
+var util = require('../../lib/common/util.js');
+
+var fakeUtil = extend({}, util);
+
+function FakeService() {
+  this.calledWith_ = arguments;
+  Service.apply(this, arguments);
+}
+
+nodeutil.inherits(FakeService, Service);
+
+function FakeFile() {
+  this.calledWith_ = arguments;
+}
+
+var requestOverride = null;
+var fakeRequest = function() {
+  return (requestOverride || util.noop).apply(this, arguments);
+};
+
+describe('Speech', function() {
+  var FILE = './audio.raw';
+  var PROJECT_ID = 'project-id';
+
+  var Speech;
+  var speech;
+
+  before(function() {
+    mockery.registerMock('request', fakeRequest);
+    mockery.registerMock('../../lib/storage/file.js', FakeFile);
+    mockery.registerMock('../../lib/common/service.js', FakeService);
+    mockery.registerMock('../../lib/common/util.js', fakeUtil);
+
+    mockery.enable({
+      useCleanCache: true,
+      warnOnUnregistered: false
+    });
+
+    Speech = require('../../lib/speech/index.js');
+  });
+
+  after(function() {
+    mockery.deregisterAll();
+    mockery.disable();
+  });
+
+  beforeEach(function() {
+    requestOverride = null;
+
+    speech = new Speech({
+      projectId: PROJECT_ID
+    });
+  });
+
+  describe('instantiation', function() {
+    it('should normalize the arguments', function() {
+      var normalizeArguments = fakeUtil.normalizeArguments;
+      var normalizeArgumentsCalled = false;
+      var fakeOptions = { projectId: PROJECT_ID };
+      var fakeContext = {};
+
+      fakeUtil.normalizeArguments = function(context, options) {
+        normalizeArgumentsCalled = true;
+        assert.strictEqual(context, fakeContext);
+        assert.strictEqual(options, fakeOptions);
+        return options;
+      };
+
+      Speech.call(fakeContext, fakeOptions);
+      assert(normalizeArgumentsCalled);
+
+      fakeUtil.normalizeArguments = normalizeArguments;
+    });
+
+    it('should inherit from Service', function() {
+      assert(speech instanceof Service);
+
+      var calledWith = speech.calledWith_[0];
+
+      var baseUrl = 'speech.googleapis.com';
+      assert.strictEqual(calledWith.baseUrl, baseUrl);
+      assert.strictEqual(calledWith.projectIdRequired, false);
+      assert.deepEqual(calledWith.scopes, [
+        'https://www.googleapis.com/auth/cloud-platform'
+      ]);
+      assert(speech.protos.Speech);
```

Comment:
It's lame, but we will want to verify that the `path`, `service` and `apiVersion` keys for both protos are set properly.